### PR TITLE
Update maintainers for Kokkos and add Kokkos 4.2.1

### PR DESCRIPTION
**Pull request by nmm0**
@crtrott @janciesko 

The Kokkos team decided that Cédric and I should maintain the Kokkos spack package.

Additionally, I added the minor release that we just tagged

### CODE REVIEW

**Review by cedricchevalier19 — APPROVED** [review]
Thanks @nmm0

**Comment by lucbv** [comment]
@nmm0 @cedricchevalier19 let us try to coordinate next time, it is nice to have both Kokkos and Kokkos Kernels in a single PR for these version updates, this way we only need to go through the whole CI, review and merge process once.

**Comment by alecbcs** [comment]
@crtrott @janciesko can you confirm you all would like to step down from being package maintainers?

**Comment by alecbcs** [comment]
@lucbv would you like to be listed as a maintainer of the package so you are notified of future updates?

**Comment by crtrott** [comment]
Yeah maybe add Luc as well. But yes those three have been doing most of the spack stuff anyway, so we decided they can be maintainers :-)

File: var/spack/repos/builtin/packages/kokkos/package.py
Comment (lucbv) [inline var/spack/repos/builtin/packages/kokkos/package.py]:
```suggestion
    maintainers("cedricchevalier19", "nmm0", "lucbv")
```

**Comment by lucbv** [comment]
Yes, adding me is no problem.
I did some of the previous PRs so that was in fact already a bit the case...

**Review by alecbcs — APPROVED** [review]
Confirmed SHAs. Thanks all for the quick turn around.

@lucbv ping me once you've rebased #42711.

**Comment by lucbv** [comment]
Okay, I'll wait for this to get merged first then will rebase on develop.
I'll ping you once we are set for a review, thanks!

**Pull request merged**
